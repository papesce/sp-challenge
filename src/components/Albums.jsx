import React, { Component } from 'react'
import Album from './Album' 
import { GridList, GridListTile } from 'material-ui/GridList';
import { withStyles } from 'material-ui/styles';
import Subheader from 'material-ui/List/ListSubheader';


const styles = theme => ({
    container: {
      flex: "1",
      display: 'flex',
      paddingLeft: 5* theme.spacing.unit,
      paddingRight:  5 * theme.spacing.unit,
      overflow: 'hidden',
      //background: theme.palette.background.paper,
    },
    gridList: {
      width: "100%",
    },
  });

export class Albums extends Component {
  render() {
     const {classes, albums, handleClick} = this.props;  
    return (
      <div className={classes.container}>    
       <GridList cellHeight={180} className={classes.gridList}>
        <GridListTile key="Subheader" cols={2} style={{ height: 'auto' }}>
          <Subheader component="div">Albums</Subheader>
        </GridListTile>
        {albums && albums.items && albums.items.map((album, index) => (
            <Album key={index} album={album} handleClick={handleClick}/>
        ))}
      </GridList>
      </div>
    )
  }
}

export default withStyles(styles)(Albums)