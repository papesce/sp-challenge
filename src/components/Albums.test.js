import React from "react";
import Albums from "./Albums";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Album tests", () => {
  let albums;
  beforeAll(() => {
    albums = {
      items: [
        {
          id: "x",
          name: "albumname",
          images: [{}, { url: "text" }],
          artists: [{ name: "name" }],
          external_urls: { spotify: "eurl" }
        }
      ]
    };
  });
  it("should ", () => {
    shallow(<Albums albums={albums} />);
  });
  it("renders correctly", () => {
    //const tree = renderer.create(<Albums albums={albums}/>)
    //    .toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
