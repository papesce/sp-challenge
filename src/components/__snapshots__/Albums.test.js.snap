// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album tests renders correctly 1`] = `
<div
  className="Albums-container-1"
>
  (
  <ul
    className="MuiGridList-root-3 Albums-gridList-2"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root-4"
      style={
        Object {
          "height": "auto",
          "padding": 2,
          "width": "100%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile-5"
      >
        <div
          className="MuiListSubheader-root-8 MuiListSubheader-sticky-12"
        >
          Albums
        </div>
      </div>
    </li>
    <li
      className="MuiGridListTile-root-4 Album-tile-13"
    >
      <div
        className="MuiGridListTile-tile-5"
      >
        <img
          alt="albumname"
          src="text"
        />
        <div
          className="MuiGridListTileBar-root-14 MuiGridListTileBar-rootBottom-15 MuiGridListTileBar-rootWithSubtitle-17"
        >
          <div
            className="MuiGridListTileBar-titleWrap-18 MuiGridListTileBar-titleWrapActionRight-20"
          >
            <div
              className="MuiGridListTileBar-title-21"
            >
              albumname
            </div>
            <div
              className="MuiGridListTileBar-subtitle-22"
            >
              <span>
                by: 
                name
              </span>
            </div>
          </div>
          <div
            className=""
          >
            <a
              className="MuiButtonBase-root-34 MuiIconButton-root-25"
              href="eurl"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
            >
              <span
                className="MuiIconButton-label-31"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-36 MuiIconButton-icon-32"
                  color="rgba(255, 255, 255, 0.54)"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-37"
              />
            </a>
          </div>
        </div>
      </div>
    </li>
  </ul>
  ) }
</div>
`;
